### DISCUSSION TITLE
Help with BeamDemo_01_Texture scripts

### DISCUSSION THREAD

**Discussion by VeronicaG24** [InfinyTech3D/SofaUnity#77] [Help - Using SofaUnity] [answered]
I'm still learning how to use Unity/Sofa/plugins so please be patient with me. 😄 
I have diligently followed all the examples, tutorials, and demos available on your website (https://infinytech3d.com/sapapi-unity3d/). However, I am still unclear on how to access the scripts, particularly those written in C#. 

In particular, in the BeamDemo_01_Texture, I am interested in understanding the keyboard arrow control functionality and the catheter elongation or shortening mechanisms. Exploring these functionalities would greatly assist me in evaluating the adaptability of the plugin to my project.

Could you please provide guidance on accessing these scripts or direct me to resources where I can find scripts leveraging the SOFA plugin and its various features?

Thank you so much in advance!

**Comment by epernod** [comment, marked as answer]
Hi,

no worry, this is the good place to learn and it will help us improving the API as well as the user doc.
So regarding the BeamDemo_01_Texture, it is integrating the [BeamAdapater plugin of SOFA](https://github.com/sofa-framework/BeamAdapter) which allows to simulate complex beam models to create wire, cable or catheter behavior.
This plugin as a component that allows to deploy the device. 

In the Unity scene we use a Sofa Key Event to propagate the key binding from unity to SOFA.
![image](https://github.com/InfinyTech3D/SofaUnity/assets/21199245/c933b7c0-f03f-45c1-b25a-57a50bc5696e)

Then in a general way, all scripts related to Unity objects that represent SOFA components are in Core/Scripts and their corresponding communication API with SOFA is inside Core/Plugins/SofaUnityAPI
![image](https://github.com/InfinyTech3D/SofaUnity/assets/21199245/cc187612-a346-4fb9-834d-6595e86d0201)

Hope it helps,
Do not hesitate if you need more information.
Best regards
